clang/tools/binary-symbol-scanner/BinarySymbolScanner.py

```diff
@@ -0,0 +1,446 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+# 
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+# 
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+#--------------------------------------------------------------------------------
+
+# Description:
+# Scanner to scrub binaries for symbols and filter them by library APIs
+#
+# Version 1.2.2
+#
+## Usage:- BinarySymbolScanner.py [-h] -c "company-name" -p "product-name" [-b "path"] [-d] [-s] [-x]
+#
+# The captured functions from the binaries in the search path will be saved in
+# 'product-name.funcs.csv'. If statistics flag is turned on, function occurrence
+# histogram will be saved to 'product-name.stats.csv'.
+# 
+# 
+#  | Options/flags                                | Description                                              |
+#  | -------------------------------------------- | ---------------------------------------------------------|
+#  | -h, --help                                   | Show this help message and exit                          |
+#  | -c COMPANY_NAME, --company-name COMPANY_NAME | Company name the product belongs to.                     |
+#  | -p PROD_NAME, --product-name PROD_NAME       | Product name for which the binaries are being scanned.   |
+#  | -b BIN_DIR, --binary-dir BIN_DIR             | Application binary directory to be scanned. If the       |
+#  |                                              | option is not provided, the script will scan the         |
+#  |                                              | current directory and all child directories for binaries.| 
+#  | -d, --debug                                  | Enable debugging help.                                   |
+#  | -s, --statistics                             | Create function histograms.                              |
+#  | -x, --extend-non-intel                       | Extend the scanning to non-Intel performance libraries.  | 
+#
+
+import argparse
+import os
+import re
+import sys
+import subprocess
+from pathlib import Path
+
+# Commands to use to determine the symbols in the binaries on Windows and Linux
+if os.name == "nt":
+    os_name = "Windows"
+    command = ["dumpbin", "/imports"]
+else:
+    if os.name == "posix":
+        os_name = "Linux"
+        command = ["nm", "--demangle"]
+    else:
+        os_name = "Unknown"
+        command = ["unknown"]
```

Comment:
L68,  i think it's better to exit, if the OS is not supported, like MAC.

Reply:
Done!